### PR TITLE
YLP-215 Display confidential mode

### PR DESCRIPTION
**Pull request by franck-fourel**
Based on Zen mode branch (PR #313)
Below is an example of the display
image

todos: 
- review tooltip so that it's centered on the lock of the main blocks
- add tests
- and probably more ...

### CODE REVIEW

**Review by yvon-dblg — COMMENTED** [review]
Missing translation in locales/en/translation.json

**Review by yvon-dblg — COMMENTED** [review]
- Since d3 & React are incompatible, I would have get the image from https://material.io/resources/icons/?icon=lock&style=outline (see example patch)
- Remove duplicate icons, put them in ./branding
- Simplify the tideline code to display then confidential mode (zen mode same problem, will be a plus)
- Tooltip incomplete
- Need to cleanup a little the code (test codes in comments...)

**Comment by franck-fourel** [comment]
As we may expect events that can be much longer than usual events, here is the strategy that has been implemented to display the ToolTip for those types of events:
- up to 3 hours event: a standard tooltip is displayed with the lock icon and the message. No title. No Hours displayed. 
image
- more than 3 hours. No tooltip displayed. The lock icon and the 'Confidential Mode' message is displayed within the event block.
image

**Comment by MarineDBL** [comment]
> As we may expect events that can be much longer than usual events, here is the strategy that has been implemented to display the ToolTip for those types of events:
> 
> * up to 3 hours event: a standard tooltip is displayed with the lock icon and the message. No title. No Hours displayed.
>   image
> * more than 3 hours. No tooltip displayed. The lock icon and the 'Confidential Mode' message is displayed within the event block.
>   image

Why is the lock green in the tooltip? Can we keep it black for design consistency?

**Comment by franck-fourel** [comment]
> Why is the lock green in the tooltip? Can we keep it black for design consistency?

it's green because it was designed that way ;-) 
but we can have it black.

**Comment by MarineDBL** [comment]
> > Why is the lock green in the tooltip? Can we keep it black for design consistency?
> 
> it's green because it was designed that way ;-)
> but we can have it black.

Yes; confirmed with Cyril. We can "paint it black"

File: packages/viz/test/components/daily/ConfidentialTooltip.test.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
+/*
+ * == BSD2 LICENSE ==
+ * Copyright (c) 2016, Tidepool Project
```
Comment (yvon-dblg) [inline packages/viz/test/components/daily/ConfidentialTooltip.test.js:3]:
```suggestion
 * Copyright (c) 2020, Diabeloop
```

File: packages/viz/src/components/daily/confidentialtooltip/ConfidentialTooltip.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
+/*
+ * == BSD2 LICENSE ==
+ * Copyright (c) 2016, Tidepool Project
```
Comment (yvon-dblg) [inline packages/viz/src/components/daily/confidentialtooltip/ConfidentialTooltip.js:3]:
```suggestion
 * Copyright (c) 2020, Diabeloop
```

Comment (yvon-dblg) [inline packages/viz/src/components/daily/confidentialtooltip/ConfidentialTooltip.js:3]:
```suggestion
 * Copyright (c) 2020, Diabeloop
```

File: packages/viz/test/components/daily/ConfidentialTooltip.test.js (hunk 1 above)
Comment (yvon-dblg) [inline packages/viz/test/components/daily/ConfidentialTooltip.test.js:3]:
```suggestion
 * Copyright (c) 2020, Diabeloop
```

File: packages/viz/src/components/daily/confidentialtooltip/ConfidentialTooltip.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+import Tooltip from '../../common/tooltips/Tooltip';
+import colors from '../../../styles/colors.css';
+import styles from './ConfidentialTooltip.css';
+
+import {Grid} from '@material-ui/core';
+import LockIcon from '@material-ui/icons/LockOutlined';
+
+const t = i18next.t.bind(i18next);
+
+class ConfidentialTooltip extends React.Component {
```
Comment (yvon-dblg) [inline packages/viz/src/components/daily/confidentialtooltip/ConfidentialTooltip.js:31]: Could have been a function, but acceptable as a class.

File: packages/viz/src/components/daily/confidentialtooltip/ConfidentialTooltip.css [hunk 4]
```diff
@@ -0,0 +1,35 @@
+/*
+ * == BSD2 LICENSE ==
+ * Copyright (c) 2016, Tidepool Project
```
Comment (yvon-dblg) [inline packages/viz/src/components/daily/confidentialtooltip/ConfidentialTooltip.css:3]:
```suggestion
 * Copyright (c) 2020, Diabeloop
```

File: packages/tideline/package.json [hunk 5]
```diff
@@ -33,6 +33,7 @@
     "react-dom": "^16.13.1",
     "react-sizeme": "2.6.12",
     "util": "^0.12.3",
-    "sundial": "file:../sundial"
+    "sundial": "file:../sundial",
+    "@material-ui/icons": "^4.9.1"
```
Comment (yvon-dblg) [inline packages/tideline/package.json:37]:
```suggestion
    "sundial": "file:../sundial"
```
Not useful anymore

File: packages/tideline/js/plot/confidentialModeEvent.js [hunk 6]
```diff
@@ -0,0 +1,119 @@
+/*
+ * == BSD2 LICENSE ==
+ * Copyright (c) 2014, Tidepool Project
```
Comment (yvon-dblg) [inline packages/tideline/js/plot/confidentialModeEvent.js:3]:
```suggestion
 * Copyright (c) 2020, Diabeloop
```

File: packages/tideline/js/plot/confidentialModeEvent.js [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+const lockIcon = require('lock.svg');
+const utils = require('./util/utils');
+
+const t = i18next.t.bind(i18next);
+
+module.exports = function (pool, options = {}) {
+  const height = pool.height() - 2;
+  const imageSize = 24;
+  // 3 hours max for the tooltip
+  const MaxSizeWithTooltip = 1000 * 60 * 60 * 3;
```
Comment (yvon-dblg) [inline packages/tideline/js/plot/confidentialModeEvent.js:32]:
```suggestion
  const maxSizeWithTooltip = 1000 * 60 * 60 * 3;
```

File: packages/tideline/js/plot/confidentialModeEvent.js [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+
+  const opts = _.cloneDeep(options);
+  opts.xScale = pool.xScale().copy();
+  const poolId = pool.id();
+
+  const xPos = (d) => utils.xPos(d, opts);
+  const calculateWidth = (d) => utils.calculateWidth(d, opts);
+  const displayTooltip = (d) => (utils.getDuration(d).duration < MaxSizeWithTooltip);
```
Comment (yvon-dblg) [inline packages/tideline/js/plot/confidentialModeEvent.js:40]:
```suggestion
  const displayTooltip = (d) => (utils.getDuration(d).duration < maxSizeWithTooltip);
```

File: packages/tideline/js/plot/confidentialModeEvent.js [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+            class: 'd3-confidential-text',
+            id: (d) => `${poolId}_confidential_lock_${d.id}`,
+          });
+
+      events.exit().remove();
+
+      // tooltips
+      selection.selectAll('.d3-confidential-group').on('mouseover', function (d) {
+        const {duration} = utils.getDuration(d);
+        if ( duration < MaxSizeWithTooltip) {
```
Comment (yvon-dblg) [inline packages/tideline/js/plot/confidentialModeEvent.js:87]:
```suggestion
        if ( duration < maxSizeWithTooltip) {
```

**Review by yvon-dblg — CHANGES_REQUESTED** [review]
Work as expected 😃 
Few comments though:
- Remove unuse image files like 
  - packages/tideline/img/lock.png
  -  packages/viz/src/images/lock.png
- Green tooltip icon
- Missing changelog

File: packages/viz/src/components/daily/confidentialtooltip/ConfidentialTooltip.js [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+
+  renderConfidential(c) {
+    return <Grid container direction="row" alignItems="center" justify="center">
+      <Grid item> 
+      <LockIcon className={styles.icon} /> 
+      </Grid>
+      <Grid item> {t('Confidential Mode')} </Grid>
```
Comment (yvon-dblg) [inline packages/viz/src/components/daily/confidentialtooltip/ConfidentialTooltip.js:38]:
```suggestion
      <Grid item>{t('Confidential Mode')}</Grid>
```

**Review by yvon-dblg — APPROVED** [review]
LGTM

**Pull request merged**